libfwupdplugin/fu-usb-device.c

```diff
@@ -452,7 +452,9 @@ fu_usb_device_open_internal(FuUsbDevice *self, GError **error)
 			return FALSE;
 		}
 		fd = fu_io_channel_unix_get_fd(io_channel);
+#ifndef __FreeBSD__
 		rc = libusb_wrap_sys_device(usb_ctx, fd, &priv->handle);
+#endif
```

Comment:
I think you need to g_set_error() something and return FALSE if you're here -- otherwise we've just got a random fd floating around and no indication it's not going to work...

Reply:
ok done